packages/gestalt/src/SheetMobile.js

```diff
@@ -0,0 +1,183 @@
+   */
+  showDismissButton?: boolean,
+  /**
+   * Subheading for SheetMobile. See the [header variant](https://gestalt.pinterest.systems/web/sheetmobile#Header) for more info.
+   */
+  subHeading?: string,
+  /**
+   * Sets the SheetMobile's height. See the [size variant](https://gestalt.pinterest.systems/web/sheetmobile#Size) for more info.
+   */
+  size?: 'default' | 'full' | 'auto',
+|};
+
+/**
+ * [SheetMobile](https://gestalt.pinterest.systems/web/sheetmobile) is a mobile only component. It is not used in desktop experiences.
+ *
+ * SheetMobile is a supplementary container that sits on top of the screen’s primary content and can be dismissed in order to interact with the underlying content. Sheets can contain a wide variety of information and layouts, including menu items, actions, and supplemental content.
+ *
+ *
+ * ![SheetMobile light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/SheetMobile.spec.mjs-snapshots/SheetMobile-chromium-darwin.png)
+ * ![SheetMobile dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/SheetMobile-dark.spec.mjs-snapshots/SheetMobile-dark-chromium-darwin.png)
+ *
+ */
+function SheetMobile({
+  accessibilityLabel,
+  align = 'start',
+  backIconButton,
+  children,
+  closeOnOutsideClick = true,
+  forwardIconButton,
+  onAnimationEnd,
+  onDismiss,
+  footer,
+  padding,
+  primaryAction,
+  heading,
+  role,
+  showDismissButton = true,
+  subHeading,
+  size = 'default',
+}: Props): Node {
+  const deviceType = useDeviceType();
+  const isMobile = deviceType === 'mobile';
+
+  if (!isMobile) return null;
+
+  if (size === 'full')
+    return (
+      <FullPage
+        accessibilityLabel={accessibilityLabel}
+        align={align}
+        backIconButton={backIconButton}
+        forwardIconButton={forwardIconButton}
+        footer={footer}
+        heading={heading}
+        onDismiss={onDismiss}
+        padding={padding}
+        primaryAction={primaryAction}
+        role={role}
+        showDismissButton
+        subHeading={subHeading}
+      >
+        {children}
+      </FullPage>
+    );
+
+  if (['default', 'auto'].includes(size))
+    return (
+      <AnimationProvider>
+        <PartialPage
+          accessibilityLabel={accessibilityLabel}
+          align={align}
+          backIconButton={backIconButton}
+          closeOnOutsideClick={closeOnOutsideClick}
+          forwardIconButton={forwardIconButton}
+          onAnimationEnd={onAnimationEnd}
+          onDismiss={onDismiss}
+          footer={footer}
+          heading={heading}
+          padding={padding}
+          primaryAction={primaryAction}
+          role={role}
+          showDismissButton={showDismissButton}
+          subHeading={subHeading}
+          size={size}
+        >
+          {children}
+        </PartialPage>
+      </AnimationProvider>
+    );
+
+  return null;
```

Comment:
do we return null, or at this point throw an error with the reason why?

Reply:
DONE

Though we could just not return, but it doesn't hurt